File: packages/vura-io/src/metrics.ts
export class MetricsManager {
    public track(name: string, value: number, step?: number | null): void {
        const payload = { type: 'vura_metric', name, value, step: step ?? null, timestamp: Date.now() };
        process.stderr.write(JSON.stringify(payload) + '\n');
    }

    public log(message: string, level: string = 'INFO'): void {
        const payload = { type: 'vura_log', level: level.toUpperCase(), message, timestamp: Date.now() };
        console.log(`[${level.toUpperCase()}] ${message}`);
    }

    public preview(name: string, sample: any): void {
        const payload = { type: 'vura_preview', name, sample, timestamp: Date.now() };
        process.stderr.write(JSON.stringify(payload) + '\n');
    }
}

export const metrics = new MetricsManager();

File: packages/core/vura-io-py/vura/io/metrics.py
import json
import sys
import time

class MetricsManager:
    def track(self, name, value, step=None):
        payload = {"type": "vura_metric", "name": name, "value": value, "step": step, "timestamp": time.time()}
        print(json.dumps(payload), file=sys.stderr)

    def log(self, message, level="INFO"):
        print(f"[{level.upper()}] {message}")

    def preview(self, name, sample):
        payload = {"type": "vura_preview", "name": name, "sample": sample, "timestamp": time.time()}
        print(json.dumps(payload), file=sys.stderr)

metrics = MetricsManager()
